idaes_examples/notebooks/docs/unit_models/custom_unit_models/liquid_extraction.py

```diff
@@ -0,0 +1,596 @@
+            & self.liquid_phase.properties_out.component_list
+        )
+
+        # Get units for unit conversion
+        aunits = self.config.aqueous_property_package.get_metadata().get_derived_units
+        lunits = self.config.liquid_property_package.get_metadata().get_derived_units
+        flow_basis = self.aqueous_phase.properties_out[t_init].get_material_flow_basis()
+
+        if flow_basis == MaterialFlowBasis.mass:
+            fb = "flow_mass"
+        elif flow_basis == MaterialFlowBasis.molar:
+            fb = "flow_mole"
+        else:
+            raise ConfigurationError(
+                f"{self.name} Liquid-Liquid Extractor only supports mass "
+                f"basis for MaterialFlowBasis."
+            )
+
+        # Material balances
+        def rule_material_aq_balance(self, t, j):
+            if j in common_comps:
+                return self.aqueous_phase.mass_transfer_term[
+                    t, "Aq", j
+                ] == -self.liquid_phase.config.property_package.diffusion_factor[j] * (
+                    self.aqueous_phase.properties_in[t].get_mass_comp(j)
+                    / self.aqueous_phase.properties_in[t].get_flow_rate()
+                )
+            elif j in self.liquid_phase.properties_out.component_list:
+                # No mass transfer term
+                # Set Liquid flowrate to an arbitary small value
+                return self.liquid_phase.mass_transfer_term[t, "Liq", j] == 0 * lunits(
+                    fb
+                )
+            elif j in self.aqueous_phase.properties_out.component_list:
+                # No mass transfer term
+                # Set aqueous flowrate to an arbitary small value
+                return self.aqueous_phase.mass_transfer_term[t, "Aq", j] == 0 * aunits(
+                    fb
+                )
+
+        self.material_aq_balance = Constraint(
+            self.flowsheet().time,
+            self.aqueous_phase.properties_out.component_list,
+            rule=rule_material_aq_balance,
+            doc="Unit level material balances for Aq",
+        )
+
+        def rule_material_liq_balance(self, t, j):
+            print(t)
+            if j in common_comps:
+                return (
+                    self.liquid_phase.mass_transfer_term[t, "Liq", j]
+                    == -self.aqueous_phase.mass_transfer_term[t, "Aq", j]
+                )
+            else:
+                # No mass transfer term
+                # Set Liquid flowrate to an arbitary small value
+                return self.liquid_phase.mass_transfer_term[t, "Liq", j] == 0 * aunits(
+                    fb
+                )
+
+        self.material_liq_balance = Constraint(
+            self.flowsheet().time,
+            self.liquid_phase.properties_out.component_list,
+            rule=rule_material_liq_balance,
+            doc="Unit level material balances Liq",
+        )
+
+    def initialize_build(
```

Comment:
This way of doing initialization is being deprecated. You should instead build an Initializer (which the MSContactor already has for you).